docs/5_mobile_native_app_generation.md

```diff
@@ -0,0 +1,1340 @@
+3. **Feature Validation**: User reviews implemented features in running app
+4. **Satisfaction Check**: User determines if features meet requirements
+   - **If satisfied**: End phase cycle
+   - **If refinement needed**: Collect feedback and repeat Design/Iterate phase
+
+### Phase 1: Plan
+
+**Objective**: Establish environment and create functioning skeletal mobile app project.
+
+#### Environment Setup
+
+- Validate required development tools for chosen mobile platform using `@salesforce/lwc-dev-mobile-core` CLI plugin
+- Leverage existing `sf force lightning local setup` command environment checks
+- _Note: CLI plugin requires updates to support structured JSON output via `--json` flag_
+
+#### Template Selection
+
+- Determine optimal `forceios` or `forcedroid` project template based on user requirements
+- Templates sourced from existing SalesforceMobileSDK-Templates repo or new template collection
+- **CLI Enhancements Required**:
+  - Support for template collection URIs in `forceios` and `forcedroid`
+  - Collection-level metadata for template registry/directory
+  - Template-specific metadata for self-describing projects
+
+#### Template Metadata System
+
+- **Collection Metadata**: Descriptive directory enabling LLM template selection
+- **Template Metadata**: Rich project information including:
+  - Nature of implemented features
+  - Design and implementation considerations
+- **Local Access**: All templates and metadata available locally per documentation repository approach
+- _Note: Metadata system design details TBD_
+
+#### Connected App Configuration
+
+- Gather required Connected App Client ID and Callback URI
+- Essential inputs for baseline mobile app project creation
+
+#### Project Creation and Setup
+
+- **Template-Based Project Generation**: Use `forceios`/`forcedroid` CLI tools to generate boilerplate app projects from Mobile SDK templates through keyword substitution
+- **Connected App Integration**: CLI tools configure OAuth parameters in generated project files using provided Connected App credentials
+- **Dependency Configuration**: Generated projects include properly configured CocoaPods/Swift Package Manager (iOS) or Gradle (Android) dependencies
+- **Ready-to-Build Foundation**: CLI output creates immediately buildable projects using native platform build systems
+
+### Phase 2: Design/Iterate
+
+**Objective**: Transform user requirements into working features through iterative development cycles with continuous validation.
+
+#### Pre-Conditions
+
+- List of user requirements from initial utterance
+- Functioning mobile app project based on selected Mobile SDK template that builds and deploys successfully
+- User can login to launched app on virtual devices
+
+#### Core Facility: Changelog System
+
+- **Iteration Tracking**: Create sequential Markdown documents (`changelog-1.md`, `changelog-2.md`, etc.) in `changelog/` folder within mobile app project
```

Comment:
for completion sake we should also specify where the original design is located. also we need to figure out when do we iterated on a design doc vs when do we create a new changelog file? Is it commit dependent? Feature dependent?, etc